enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -0,0 +1,806 @@
+5. The service end-user connects to the host name specified in the HTTPRoute from Step 4 and receives a response from the service that the HTTPRoute targets.
+
+### API Extensions
+
+This enhancement imports the v1beta1 versions of the following CRDs into OpenShift:
+
+* [GatewayClass](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/).
+* [Gateway](https://gateway-api.sigs.k8s.io/api-types/gateway/).
+* [HTTPRoute](https://gateway-api.sigs.k8s.io/api-types/httproute/).
+
+Before this enhancement, DNSRecord CRs were not reconciled unless they were in
+the "openshift-ingress-operator" namespace and had an associated
+IngressController.  This enhancement enables reconciliation of DNSRecord CRs in
+the "openshift-ingress" namespace.
+
+Tentatively, this enhancement may define an additional CRD for specifying
+configuration such as the following:
+
+* Whether Gateway API/ingress is enabled.
+* Whether Istio API/mesh is enabled.
+* Whether to create a default Gateway.
+* Istio parameters, such as profile and addons (Jaeger, Kiali, Grafana, Prometheus).
+* OSSM automatic deployments.
+
+See [Open Questions](#open-questions) for some related discussion.
+
+### Implementation Details/Notes/Constraints
+
+We have several constraints related to design and release-timing constraints to consider.
+
+#### Ownership of the ServiceMeshControlPlane
+
+OpenShift Service Mesh is configured through the ServiceMeshControlPlane (SMCP)
+CR, both for ingress use-cases and for mesh use-cases.  [The existing
+installation procedure for Service Mesh](https://docs.openshift.com/container-platform/4.11/service_mesh/v2x/ossm-create-smcp.html#ossm-control-plane-deploy-cli_ossm-create-smcp) instructs the reader to create a
+SMCP named "basic" in the "istio-system" namespace.  To ensure backwards
+compatibility, the Ingress Operator must do one of the following:
+
+* Use the same name and namespace: "istio-system/basic".
+* Migrate an existing SMCP from "istio-system" to "openshift-ingress".
+* Provide an API to tell the Ingress Operator what SMCP to use.
+
+Placement, possible migration, and ownership of the SMCP need to be clarified in
+this enhancement.  This is an open design question.
+
+#### Managing the ServiceMeshMemberRoll
+
+OSSM 2.3 only watches resources in namespaces specified in an
+ServiceMeshMemberRoll CR associated with the ServiceMeshControlPlane; see
+[OSSM-924](https://issues.redhat.com/browse/OSSM-924).  We hope that this limitation will be removed from OSSM in time
+for this enhancement; this is an open release-planning question.  Otherwise, the
+Ingress Operator or cluster admin must add each namespace with HTTPRoute objects
+that Istio should reconcile to the ServiceMeshMemberRoll, and the approach we
+take and ownership of ServiceMeshMemberRoll need to be clarified in this
+enhancement; this is an open design question.
+
+#### Gateway API v1alpha1 CRDs
+
+Istio fails to become ready if the expected CRDs do not all exist.  OSSM 2.3 is
```

Comment:
Here this paragraph states essentially "OSSM 2.3 won't start without v1alpha1 Gateway API CRDS":
1. Did you mean v1alpha2?
2. Are you saying that OSSM v2.3 only supports v1alpha1? 

OSSM 2.3 has Istio 1.14.3, which has Gateway API v0.4.1, which has v1alpha2 CRDs. It won't start without v1alpha1 CRDs? Just started a OSSM demo with v1alpha2 and v1beta1 versions (v0.5.0 with experimental on, so you get v1alpha2).

Reply:
I meant v1alpha2, sorry.  I'll fix it.